src/test/java/com/datadog/api/World.java

```diff
@@ -0,0 +1,226 @@
+        // client.setUnstableOperationEnabled(operationId, true)
+        clientClass.getMethod("setUnstableOperationEnabled", String.class, boolean.class).invoke(client, operationId,
+                true);
+    }
+
+    public void configureApiKeys(Map<String, String> secrets)
+            throws java.lang.reflect.InvocationTargetException, java.lang.IllegalAccessException,
+            java.lang.InstantiationException, java.lang.NoSuchMethodException, java.lang.ClassNotFoundException {
+        // client.configureApiKeys(secrets)
+        clientClass.getMethod("configureApiKeys", Map.class).invoke(client, secrets);
+    }
+
+    public void newRequest(String methodName) {
+        requestParams = new HashMap<String, Object>();
+        for (Method method : apiClass.getMethods()) {
+            if (method.getName().equals(methodName)) {
+                requestBuilder = method;
+                requestClass = method.getReturnType();
+                break;
+            }
+        }
+    }
+
+    public Method getRequestUndo() throws java.lang.ClassNotFoundException, java.lang.NoSuchMethodException {
+        String actionName = requestBuilder.getName();
+        if (Stream.of(noUndo).anyMatch(u -> actionName.startsWith(u))) {
+            return null;
+        }
+
+        Class<?> undoClass = Class.forName(apiClass.getPackage().getName() + ".Undo");
+        Method dataMethod = responseClass.getMethod("getData");
+        return undoClass.getMethod(actionName, apiClass, dataMethod.getReturnType());
+    }
+
+    public void sendRequest()
+            throws java.lang.ClassNotFoundException, java.lang.IllegalAccessException, java.lang.NoSuchMethodException,
+            java.lang.reflect.InvocationTargetException, com.fasterxml.jackson.core.JsonProcessingException {
+        Object request;
+        if (requestBuilder.getParameterCount() > 0) {
+            Object[] parameters = new Object[requestBuilder.getParameterCount()];
+            request = requestBuilder.invoke(api, parameters);
+        } else {
+            request = requestBuilder.invoke(api);
+        }
+
+        for (Field f : requestClass.getDeclaredFields()) {
+            if (requestParams.containsKey(f.getName())) {
+                f.setAccessible(true);
+                f.set(request, requestParams.get(f.getName()));
+            }
+        }
+
+        Method responseMethod = requestClass.getMethod("executeWithHttpInfo");
+        responseClass = responseMethod.getReturnType();
+
+        Method undoMethod = getRequestUndo();
+
+        System.err.printf("\n\n%s\n\n", request.toString());
+        response = responseMethod.invoke(request);
+
+        if (undoMethod != null) {
+            // TODO append to undo list
+            undo.add(() -> {
+                Method dataMethod = responseClass.getMethod("getData");
+                undoMethod.invoke(null, apiClass.cast(api), dataMethod.invoke(response));
+                return null;
+            });
```

Comment:
we should either undo in reverse order in clientSteps.java, or prepend to the list here.
Also this isn't a TODO anymore right ?

Reply:
I have removed the TODO

Reply:
What about the reverse order thing ?